src/main/java/com/vi/migrationtool/tenantservice/ImportDiocesesFromAgencyToTenantServiceTask.java

```diff
@@ -0,0 +1,70 @@
+package com.vi.migrationtool.tenantservice;
+
+import com.vi.migrationtool.common.MigrationTasks;
+import com.vi.migrationtool.config.BeanAwareSpringLiquibase;
+import liquibase.database.Database;
+import lombok.Builder;
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+import org.springframework.jdbc.core.JdbcTemplate;
+
+@Slf4j
+public class ImportDiocesesFromAgencyToTenantServiceTask extends MigrationTasks {
+
+  private static final String CREATE_DIOCESE_TABLE =
+      "CREATE TABLE IF NOT EXISTS diocese (id BIGINT NOT NULL AUTO_INCREMENT, name VARCHAR(255) NOT NULL, PRIMARY KEY (id) UNIQUE (name));";
+  private static final String dioceseTableName = "diocese";
+
+  @Builder
+  static class Diocese {
+    private String name;
+  }
+
+  @Override
+  public void execute(Database database) {
+    log.info("Starting import of dioceses from agency to tenant");
+    val agencyJdbcTemplate =
+        BeanAwareSpringLiquibase.getNamedBean("agencyServiceJdbcTemplate", JdbcTemplate.class);
+
+    var dioceseTableExist = doesTableExist(agencyJdbcTemplate, "agencyservice");
+    if (!dioceseTableExist) {
+      log.info("Diocese table does not exist in agency service, skipping import");
+      return;
+    }
+    dioceseTableExist = doesTableExist(agencyJdbcTemplate, "tenantservice");
```

Comment:
method should be named to dioceseTableExists or "diocese" should be passed as additional method param